### PR TITLE
Block launching the application with Java 9

### PR DESCRIPTION
**Pull request by AustinShalit**

### CODE REVIEW

**Comment by SamCarlberg** [comment]
What happens when running OutlineViewer with Java 9?

File: src/main/java/edu/wpi/first/outlineviewer/OutlineViewer.java [hunk 1]
```diff
@@ -15,9 +16,24 @@
...
+  private static final String JRE_VERSION = System.getProperty("java.specification.version");
 
   @Override
   public void start(Stage primaryStage) throws IOException {
+    if (!REQUIRED_JRE_VERSION.equals(JRE_VERSION)) {
+      Alert invalidVersionAlert = new Alert(Alert.AlertType.ERROR);
+      invalidVersionAlert.setHeaderText("Invalid JRE Version!");
+      invalidVersionAlert.setContentText(
+          String.format("You are using an unsupported Java version: %s!  "
+              + "Please download Java %s.",
```
Comment (PeterJohnson) [inline src/main/java/edu/wpi/first/outlineviewer/OutlineViewer.java]: This is going to say "Java 1.8", correct?  It's commonly known as Java 8, I don't know if Oracle calls it 1.8 anywhere.

Comment (AustinShalit) [reply src/main/java/edu/wpi/first/outlineviewer/OutlineViewer.java]: I'll hardcode it to say 8.

**Comment by AustinShalit** [comment]
>What happens when running OutlineViewer with Java 9?

ControlsFX does not support Java 8 & Java 9 at the same time.  The only long term solution I can think of is removing that dependency.  This will work for now though.

**Comment by JLLeitschuh** [comment]
Why doesn't outline viewer get published as an application with a bundled version of the JDK?

File: src/main/java/edu/wpi/first/outlineviewer/OutlineViewer.java [hunk 2]
```diff
@@ -18,6 +19,18 @@
...
+    if (!"1.8".equals(System.getProperty("java.specification.version"))) {
+      Alert invalidVersionAlert = new Alert(Alert.AlertType.ERROR);
+      invalidVersionAlert.setHeaderText("Invalid JRE Version!");
+      invalidVersionAlert.setContentText(
+          String.format("You are using an unsupported Java version: %s!  "
+              + "Please download Java 8.",
+              System.getProperty("java.version")));
+      invalidVersionAlert.showAndWait();
+
+      return;
```
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/first/outlineviewer/OutlineViewer.java:31]: Quit after this?

**Comment by AustinShalit** [comment]
@JLLeitschuh If we did that for all of our tools apps using javafx, the size of the eclipse plugins would skyrocket.

**Comment by SamCarlberg** [comment]
Bundling a ~175MB JRE for a 13MB application is way overkill, especially since you need JDK8 installed for FRC anyway

**Comment by JLLeitschuh** [comment]
Makes sense.

**Pull request merged**